File: user-provider-9001/src/main/java/cn/liangqinghai/study/spring/cloud/provider/user/controller/UserController.java
package cn.liangqinghai.study.spring.cloud.provider.user.controller;

import cn.liangqinghai.study.spring.cloud.common.dto.ResultDto;
import cn.liangqinghai.study.spring.cloud.common.models.User;
import cn.liangqinghai.study.spring.cloud.common.repos.UserRepos;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * @author LiangQinghai
 * @title UserController
 * @projectName spring-cloud-study
 * @description
 * @date 2020/8/3 14:41
 */
@RestController
@RequestMapping("/cloud/user")
public class UserController {

    @GetMapping("/list")
    public ResultDto<List<User>> list() {

        return ResultDto.ok(UserRepos.users);

    }

}

File: project-web-auth/src/main/kotlin/cn/liangqinghai/study/spring/cloud/project/auth/controller/UserController.kt
package cn.liangqinghai.study.spring.cloud.project.auth.controller

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.security.Principal

@RestController
@RequestMapping("/oauth")
class UserController {

    @RequestMapping("/user")
    fun user(user: Principal): Principal {

        return user

    }

}